### ISSUE TITLE
Low TPU utilization and RAM Memory Leak

### ISSUE THREAD

**Issue by narendra4** [pytorch/xla#3104]
## ❓ Questions and Help
I am trying to test run pytorch xla with multiprocessing on kaggle competition.  
While the training proceeds I am getting RAM oom and also the TPU consumption is very low.  I tried to figure out for the xla lowering and the cpu context changes, but not able to understand the issue.

CPU is gettting utilized very high and RAM is consuming as the iterations progress.

Training Size: (343146, 28)

class TestModel(nn.Module):

    def __init__(self):
        super().__init__()
        hsize=256
        self.pre_bn=nn.BatchNorm1d(config.num_buckets , config.num_features)
        self.gru=nn.GRU(config.num_features, hsize, 2, dropout=0.2)
        
    def forward(self, x):
        x=self.pre_bn(x)
        h, _ =self.gru(x)
        return h

    

train_dataset=OptiverDataset(train_df)

def test_run(rank, FLAGS):

    torch.set_default_tensor_type('torch.FloatTensor')

    train_sampler=torch.utils.data.distributed.DistributedSampler(
        train_dataset,
        num_replicas=8,
        rank=rank,
        shuffle=True
    )
    
    train_data_loader = torch.utils.data.DataLoader(
        train_dataset,
        batch_size=128,
        sampler=train_sampler,
        drop_last=True,
        num_workers=1,
        pin_memory=True
    )
    
    
    device = xm.xla_device()
    model=TestModel().to(device)
    optimizer=torch.optim.AdamW(model.parameters(), lr=8e-5, weight_decay=1e-5)
    
    train_para_loader=pl.MpDeviceLoader(train_data_loader, device)
    for i, data in enumerate(train_para_loader):
        (X, yprimary, yask1, ybid1, ywap1)=(data['X'], data['yprimary'],data['yask1'], data['ybid1'], data['ywap1'])

        X=X.to(device)
        yprimary=yprimary.to(device)
        yask1=yask1.to(device)
        ybid1=ybid1.to(device)
        ywap1=ywap1.to(device)
        
        with torch.no_grad():
            out=model(X)
        if i%100==0:
            xm.master_print("Iteration:{} - out_size:{}".format(i, len(out)))
            
FLAGS={}
xmp.spawn(test_run, args=(FLAGS,), nprocs=8, start_method="fork")

Any help is appreciated.
Thanks
